doc/sphinx/proofs/writing-proofs/rewriting.rst

```diff
@@ -90,11 +93,66 @@ Rewriting with Leibniz and setoid equality
      any of them can be rewritten.  If not specified, only the first occurrence
      in the conclusion is replaced.
 
-     If :n:`at @occs_nums` is specified, rewriting is always done with
-     :ref:`setoid rewriting <generalizedrewriting>`, even for Leibniz’s equality,
-     which means that you must `Require Setoid` to use that form.
-     However, note that :tacn:`rewrite` (even when using setoid rewriting) and
-     :tacn:`setoid_rewrite` don't behave identically (as already mentioned above).
+     .. note::
+
+        If :n:`at @occs_nums` is specified, rewriting is always done
+        with :ref:`setoid rewriting <generalizedrewriting>`, even for
+        Leibniz equality, which means that you must `Require
+        Setoid` to use that form.  However, note that :tacn:`rewrite`
+        (even when using setoid rewriting) and :tacn:`setoid_rewrite`
+        don't behave identically (as already mentioned above).
+
+     .. note::
+
+        For each selected hypothesis (and the conclusion),
+        :tacn:`rewrite` instantiates the rewriting lemma first
+        before taking the `at` clause into account, which limits the
+        number of occurrences that can be rewritten.  You can force
+        :tacn:`rewrite` to instantiate the rewriting lemma differently
+        by using a `with` clause (see :n:`@one_term_with_bindings`) or
+        you can use :tacn:`setoid_rewrite`, which considers each
+        occurrence separately.
```

Comment:
```suggestion
        For each selected hypothesis and/or the conclusion,
        :tacn:`rewrite` finds the first matching subterm in depth-first search
        order and binds the free variables of the lemma.  Then, if the `at` clause
        is specified, only subterms with the same bindings for the free variables
        will be counted in determining the occurrence number.  This limits
        which matches of the lemma can be rewritten.  You can get around
        this limitation by explicitly specifying how some or all of the free
        variables are bound
        by using a `with` clause (see :n:`@one_term_with_bindings`).
        You can also use :tacn:`setoid_rewrite`, which does the matching before
        binding the free variables of the lemma.
```

As I commented elsewhere, "instantiates" is undefined and IMHO unclear.  "instantiate" means "to represent (an abstraction) by a concrete instance" (Merriam-Webster) or the less clear "represent as or by an instance" (Oxford).  https://www.dictionary.com/browse/instantiation does have the intended meaning: "the process of deriving an individual statement from a general one by replacing the variable with a name or other referring expression"--this meaning is new to me, perhaps should be a defined term.  I am accustomed to "instantiation" meaning to create an object belonging to a class (aka create an "instance").

Reply:
"bind" is a good replacement, probably more accurate and specific.